libtiledbsoma/src/utils/arrow_adapter.h

```diff
@@ -229,6 +229,24 @@ class ArrowAdapter {
     static std::unique_ptr<ArrowSchema> arrow_schema_from_tiledb_array(
         std::shared_ptr<Context> ctx, std::shared_ptr<Array> tiledb_array);
 
+    /**
+     * @brief Create a an ArrowSchema from TileDB Dimension
+     *
+     * @return ArrowSchema
+     */
+    static std::unique_ptr<ArrowSchema> arrow_schema_from_tiledb_dimension(
+        const Dimension& dimension);
+
+    /**
+     * @brief Create a an ArrowSchema from TileDB Attribute
+     *
+     * @return ArrowSchema
+     */
+    static std::unique_ptr<ArrowSchema> arrow_schema_from_tiledb_attribute(
+        Attribute& attribute,
+        std::shared_ptr<Context> ctx,
+        std::shared_ptr<Array> tiledb_array);
```

Comment:
Looks like you can change the implementation so these are passed as const references instead of pointers without too much pain.